libs/ssl-config/src/test/java/org/elasticsearch/common/ssl/KeyStoreUtilTests.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.ssl;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.nio.file.Path;
+import java.security.KeyStore;
+import java.util.Collections;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.equalTo;
+
+public class KeyStoreUtilTests extends ESTestCase {
+    private static final char[] P12_PASS = "p12-pass".toCharArray();
+
+    public void testFilter() throws Exception {
+        assumeFalse("Can't use PKCS#12 keystores in a FIPS JVM", inFipsJvm());
+
+        final Path p12 = getDataPath("/certs/cert-all/certs.p12");
+        final KeyStore original = KeyStoreUtil.readKeyStore(p12, "PKCS12", P12_PASS);
+
+        // No-op filter
+        final KeyStore clone = KeyStoreUtil.filter(KeyStoreUtil.readKeyStore(p12, "PKCS12", P12_PASS), entry -> true);
+        assertThat(Collections.list(clone.aliases()), containsInAnyOrder("cert1", "cert2"));
+        assertSameEntry(original, clone, "cert1", P12_PASS);
+        assertSameEntry(original, clone, "cert2", P12_PASS);
+
+        // Filter by alias
+        final KeyStore cert1 = KeyStoreUtil.filter(
+            KeyStoreUtil.readKeyStore(p12, "PKCS12", P12_PASS),
+            entry -> entry.getAlias().equals("cert1")
+        );
+        assertThat(Collections.list(cert1.aliases()), containsInAnyOrder("cert1"));
```

Comment:
Should we assert "cert2" aliases are not there? (if I understand the test correctly)

Reply:
`containsInAnyOrder` is a bit misleading - it's more acurately thought of as `equalsIgnoringOrder`.